File: takengine/takengine/src/main/java/com/atakmap/map/projection/Datum.java
package com.atakmap.map.projection;

public class Datum {

    public final static Datum WGS84 = new Datum(Ellipsoid.WGS84, 0d, 0d, 0d);

    public final double deltaX;
    public final double deltaY;
    public final double deltaZ;
    
    public final Ellipsoid referenceEllipsoid;
    
    public Datum(Ellipsoid ref, double dx, double dy, double dz) {
        this.referenceEllipsoid = ref;
        this.deltaX = dx;
        this.deltaY = dy;
        this.deltaZ = dz;
    }    
}

File: takkernel/engine/src/main/cpp/core/Datum.cpp
#include "core/Datum.h"

using namespace atakmap::core;

static Datum createWGS84() {
    Ellipsoid ellipsoid = Ellipsoid::createWGS84();
    return Datum(&ellipsoid, 0, 0, 0);
}

const Datum Datum::WGS84 = createWGS84();


Datum::Datum(const Ellipsoid *ref, const double dx, const double dy, const double dz) :
    reference(ref->semiMajorAxis, ref->flattening),
    deltaX(dx),
    deltaY(dy),
    deltaZ(dz)
{}

Datum::~Datum()
{}

